dist/fabric.js

```diff
@@ -13325,11 +13334,25 @@ fabric.PatternBrush = fabric.util.createClass(fabric.PencilBrush, /** @lends fab
      * @param {Event} e Event object fired on mouseup
      */
     __onMouseWheel: function(e) {
+      this._interruptDrawing();
       this._cacheTransformEventData(e);
       this._handleEvent(e, 'wheel');
       this._resetTransformEventData();
     },
 
+    /**
+     * Method to call if moving the drawing canvas (e.g. on a scroll event)
+     * @private
+     */
+    _interruptDrawing: function() {
+      if (this.isDrawingMode && this._isCurrentlyDrawing && this.freeDrawingBrush.interruptDrawing) {
```

Comment:
Why the need to check for the `interruptDrawing` function?

Reply:
canvas/this.freeDrawingBrush may not be the pencil_brush. And the other brush classes don't use the same path functions so I can't just make an interruptDrawing in the base_brush class.

Reply:
I had the same question. It reads as if its checking for the existence of a callback before calling the callback... and makes it seem like a bug that there are other things gated by this. 

It seems here what you are doing by having this condition "this.freeDrawingBrush.interruptDrawing" is detecting "if the current freeDrawingBrush is a Pencil brush". I guess it could also read as "is this an interruptible brush"... and if so interrupt it... otherwise leave as is. This 2nd interpretation makes a lot more sense to me (I know both are my interpretations! lol).

Anyway... I get it now... maybe just drop a comment here for clarity?

Reply:
Ok, I agree this is not super clear. I added an explicit function to the base brush to check if a brush is interruptable. 

The other things that happen along with the call to freedrawingBrush.interruptDrawing()  (like setting _isCurrentlyDrawing to false) would cause pretty weird and buggy behavior if done without the interruptDrawing() call so I wouldn't want to JUST gate the call to interruptDrawing() if the brush doesn't support it. We only use the pencil brush at the moment - so I don't know what an interruptDrawing() call to another brush would need to look like, but at least with this gating it won't inject any weird errors if we decide to add a spray-brush (for example).